Unable to import from @influxdata/influxdb-client in an ES module

I'm creating a package using typescript 4.7 with `"type": "module"` in package.json (to create a es2021 module instead of commonjs). 

__Expected Behavior:__
I should be able to import code from `@influxdata/influxdb-client` in a typescript environment.


__Current Behavior:__
The typescript compiler refuses to import code from `@influxdata/influxdb-client`:

```
src/influx.ts:1:22 - error TS7016: Could not find a declaration file for module '@influxdata/influxdb-client'. '/Users/jaap/Projects/LY/tmp2/node_modules/@influxdata/influxdb-client/dist/index.mjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/influxdata__influxdb-client` if it exists or add a new declaration (.d.ts) file containing `declare module '@influxdata/influxdb-client';`

1 import { flux } from '@influxdata/influxdb-client';
```

__Possible Solution:__

When removing the `"exports": { ... }` section from `@influxdata/influxdb-client/package.json` the code will import correctly.

__Steps/Code to Reproduce the Problem:__

1. Checkout https://github.com/q42jaap/influxdata-influxdb-client-issue
1. run `npm install`
2. run `npm run build`

__Specifications:__

- Version: @influxdata/influxdb-client 1.27.0
- Platform: node 16.15, typescript 4.7.4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 ## 1.28.0 [unreleased]
 
+### Features
+
+1. [#460](https://github.com/influxdata/influxdb-client-js/pull/460): Add type definitions for typescript 4.7+.
+
 ## 1.27.0 [2022-06-24]
 
 ### Features

--- a/packages/apis/package.json
+++ b/packages/apis/package.json
@@ -23,6 +23,7 @@
   "types": "dist/all.d.ts",
   "exports": {
     ".": {
+      "types": "./dist/all.d.ts",
       "import": "./dist/index.mjs",
       "require": "./dist/index.js",
       "deno": "./dist/index.mjs",

--- a/packages/core-browser/package.json
+++ b/packages/core-browser/package.json
@@ -16,6 +16,7 @@
   "types": "dist/all.d.ts",
   "exports": {
     ".": {
+      "types": "./dist/all.d.ts",
       "import": "./dist/index.browser.mjs",
       "require": "./dist/index.browser.js",
       "deno": "./dist/index.browser.mjs",

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -27,6 +27,7 @@
   "types": "dist/all.d.ts",
   "exports": {
     ".": {
+      "types": "./dist/all.d.ts",
       "import": "./dist/index.mjs",
       "require": "./dist/index.js",
       "deno": "./dist/index.browser.mjs",

--- a/packages/giraffe/package.json
+++ b/packages/giraffe/package.json
@@ -23,6 +23,7 @@
   "types": "dist/all.d.ts",
   "exports": {
     ".": {
+      "types": "./dist/all.d.ts",
       "import": "./dist/index.mjs",
       "require": "./dist/index.js",
       "default": "./dist/index.js"

